File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/ENIntervalConverter.java
package org.coralibre.android.sdk.internal.database.persistent;

import androidx.room.TypeConverter;

import org.coralibre.android.sdk.internal.crypto.ENInterval;

public class ENIntervalConverter {

    @TypeConverter
    public static ENInterval toENNumber(Long value) {
        return new ENInterval(value);
    }

    @TypeConverter
    public static Long toLong(ENInterval value) {
        return value.get();
    }

}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/ENIntervalConverter.kt
package org.coralibre.android.sdk.internal.database.persistent

import androidx.room.TypeConverter
import org.coralibre.android.sdk.internal.datatypes.ENInterval

object ENIntervalConverter {
    @JvmStatic
    @TypeConverter
    fun toENNumber(value: Long): ENInterval {
        return ENInterval(value)
    }

    @JvmStatic
    @TypeConverter
    fun toLong(value: ENInterval): Long {
        return value.get()
    }
}
